channels/urbdrc/client/searchman.c

```diff
@@ -219,7 +220,15 @@ USB_SEARCHMAN* searchman_new(void * urbdrc, UINT32 UsbDevice)
 	
 	searchman->strated = 0;
 	searchman->term_event = CreateEvent(NULL, TRUE, FALSE, NULL);
+	if (!searchman->term_event)
+		goto out_error_event;
 	sem_init(&searchman->sem_term, 0, 0);
 	
 	return searchman;
+
+out_error_event:
+	pthread_mutex_destroy(&searchman->mutex);
+out_error_mutex:
+	free(searchman);
+	return NULL;
```

Comment:
The problem here still might be that the return isn't checked outside after the call.
Maybe you could add the corresponding checks as well.